src/ser.rs

```diff
@@ -741,3 +762,151 @@ where
         self.end_inner()
     }
 }
+
+/// Serializing a CID correctly as DAG-CBOR.
+struct CidSerializer<'a, W>(&'a mut Serializer<W>);
+
+impl<'a, W> ser::Serializer for &'a mut CidSerializer<'a, W>
+where
+    W: Write,
+{
+    type Ok = ();
+    type Error = Error;
+
+    type SerializeSeq = ser::Impossible<Self::Ok, Self::Error>;
+    type SerializeTuple = ser::Impossible<Self::Ok, Self::Error>;
+    type SerializeTupleStruct = ser::Impossible<Self::Ok, Self::Error>;
+    type SerializeTupleVariant = ser::Impossible<Self::Ok, Self::Error>;
+    type SerializeMap = ser::Impossible<Self::Ok, Self::Error>;
+    type SerializeStruct = ser::Impossible<Self::Ok, Self::Error>;
+    type SerializeStructVariant = ser::Impossible<Self::Ok, Self::Error>;
+
+    fn serialize_bool(self, _value: bool) -> Result<Self::Ok> {
+        unreachable!();
```

Comment:
So, this should _actually_ be unreachable, but it would be nice to have an error anyways.